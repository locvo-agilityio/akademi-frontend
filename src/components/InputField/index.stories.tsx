import { StoryObj, Meta } from '@storybook/react';

import InputField from '.';

const defaultProps = {
  label: 'Email',
  name: 'email',
  placeholder: 'Email',
  onChange: () => {},
};

const meta = {
  title: 'Components/InputField',
  component: InputField,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
} satisfies Meta<typeof InputField>;

export default meta;

type Story = StoryObj<typeof InputField>;

export const Default: Story = {
  args: {
    type: 'password',
    ...defaultProps,
  },
};

export const HasError: Story = {
  args: {
    ...defaultProps,
    isError: true,
  },
};
