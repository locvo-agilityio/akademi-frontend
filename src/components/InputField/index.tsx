'use client';

import { ChangeEvent, memo } from 'react';
import {
  Input,
  InputProps,
  FormLabel,
  FormErrorMessage,
  FormControl,
} from '@chakra-ui/react';

type TInputFieldProps = Omit<InputProps, 'onChange'> & {
  isError?: boolean;
  errorMessages?: string;
  label?: string;
  onChange: (value: string) => void;
};

const InputField = ({
  isError = false,
  errorMessages = 'Default error',
  label,
  onChange,
  ...rest
}: TInputFieldProps) => {
  const handleChangeValue = (e: ChangeEvent<HTMLInputElement>): void =>
    onChange(e.target.value);

  return (
    <FormControl isInvalid={isError}>
      {label && (
        <FormLabel
          fontSize="md"
          fontWeight="semibold"
          marginInlineEnd={0}
          minW="max-content"
          color={isError ? 'red' : 'darkBlue'}
        >
          {label}
        </FormLabel>
      )}

      <Input
        py={5}
        variant="primary"
        onChange={handleChangeValue}
        {...rest}
        isInvalid={isError}
      />
      {isError && (
        <FormErrorMessage color="red" fontSize="2xs">
          {errorMessages}
        </FormErrorMessage>
      )}
    </FormControl>
  );
};

export default memo(InputField);
