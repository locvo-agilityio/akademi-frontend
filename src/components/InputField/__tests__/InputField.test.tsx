import { fireEvent, render } from '@testing-library/react';

// Components
import InputField from '..';

describe('InputField component', () => {
  const mockOnChange = jest.fn();
  const props = {
    label: 'Email',
    name: 'email',
    placeholder: 'Email',
    onChange: mockOnChange,
  };
  it('Render correctly with default props', () => {
    const { container } = render(<InputField {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('Render correctly with error message', () => {
    const { getByText } = render(
      <InputField {...props} isError errorMessages="Field is required" />,
    );
    expect(getByText('Field is required')).toBeInTheDocument();
  });

  it('should call onChange when input value is changed', async () => {
    const { getByPlaceholderText } = render(<InputField {...props} />);
    fireEvent.change(getByPlaceholderText('Email'), {
      target: { value: 'user@gmail.com' },
    });

    expect(mockOnChange).toHaveBeenCalled();
  });
});
