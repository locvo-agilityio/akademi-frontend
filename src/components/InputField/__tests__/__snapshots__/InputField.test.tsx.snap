// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component Render correctly with default props 1`] = `
<div>
  <div
    class="chakra-form-control css-0"
    role="group"
  >
    <label
      class="chakra-form__label css-1hm3o85"
      for="field-:r0:"
      id="field-:r0:-label"
    >
      Email
    </label>
    <input
      class="chakra-input css-1lx593h"
      id="field-:r0:"
      name="email"
      placeholder="Email"
    />
  </div>
</div>
`;
